src/Language/PureScript/AST/Declarations.hs

```diff
@@ -451,7 +473,7 @@ data Declaration
   -- |
   -- A value declaration (name, top-level binders, optional guard, value)
   --
-  | ValueDeclaration SourceAnn Ident NameKind [Binder] [GuardedExpr]
+  | ValueDeclaration {-# UNPACK #-} !(ValueDeclarationData [GuardedExpr])
```

Comment:
The strictness annotations on record fields primarily server to not accidentally initialize a partial record. I put the `UNPACK` here to make sure that I didn't make things worse (this should basically be the same memory layout as before)